FEATURE_PROCESS.adoc

```diff
@@ -1,5 +1,106 @@
+== The Process
+
+Feature development proceeds in five stages:
+
+=== Kickoff
+
+When a developer intends to begin work on a feature they should do the following:
+
+* If one doesn't already exist, file an issue in the relevant project JIRA, typically https://issues.redhat.com/projects/WFLY/summary[WFLY], https://issues.redhat.com/projects/WFCORE/summary[WFCORE] or https://issues.redhat.com/projects/HAL/summary[HAL].
+* File an issue with the https://github.com/orgs/wildfly/projects/7[WildFly Feature Planning project]. To do this create an issue using the https://github.com/wildfly/wildfly-proposals/issues/new?assignees=&labels=feature&projects=wildfly%2F7&template=feature-development.yaml[Feature Development issue template].
+* Draft a minimal feature analysis, based on the current https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc[design-doc-template]. This should include the expected stability level, links to the issues above, and enough Overview and User Story information to give people interested in being part of the feature team a sense of what would be involved.
+* Submit a draft analysis PR to the https://github.com/wildfly/wildfly-proposals[`wildfly-proposals`] repository.
+* Send an email to the `wildfly-dev` mail list telling people about the upcoming work and asking for volunteers to participate in the feature team.
+
+When the above tasks are complete, change the `Status` field of the WildFly Feature Planning project JIRA to `Planning`.
+
+=== Planning
+
+The Planning stage of a feature consists of <<feature-team,feature team>> formation and enough discussion of the feature among the team that they can agree on a target feature stability level and target WildFly release in which they expect to complete the feature.
+
+NOTE: The target WildFly release is just a target and is not a requirement.
+
+Update the feature analysis document with the planned stability level (if it has changed) and the GitHub ids of the feature team members.
+
+Set the `Iteration` field to the iteration for the WildFly release.
+
+When the above tasks are complete, change the `Status` field of the WildFly Feature Planning project JIRA to `In Progress`.
+
+IMPORTANT: Changes to the `Iteration` field of a WildFly Feature Planning project issue should always reflect the consensus of the feature team.
+
+=== In Progress
+
+The main work on the feature happens in this stage. The feature team works to meet the <<requirements>> that apply to the feature's target stability.
+
+When the team believes all development work on the feature is complete, change the `Status` field of the WildFly Feature Planning project JIRA to `Review`.
+
+NOTE: Developers should not move an issue to `Review` status as a means of asking other team members to do routine code review. Such intra-team coordination should be handled in some other way. Moving an issue to `Review` is a signal to WildFly release coordinators that work on the feature is about to conclude.
```

Comment:
@jamezp what did you think?

Reply:
Signoff?

Reply:
In a discussion today we agreed to drop the 'Review/Signoff' content until their is some sort of automation in place to support it, i.e. adding a stability-specific checklist to the issue description that the feature team would check off. And it would probably make more sense to do that addition when the issue moves to In Progress.

We also agreed to add a 'Ready for Merge' status, to which the feature team should move the issue once all requirements have been met. We discussed some automation around that (i.e. team checks off requirements and automation moves it when done), but that's for the future.